.github/workflows/tests_and_lint.yml

```diff
@@ -0,0 +1,45 @@
+name: Tests and lint
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v*"
+  pull_request:
+    branches:
+      - "*"
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
+    steps:
+      - name: Install Python dependencies
```

Comment:
nit: for debugging later on,

`      - name: Set up Python ${{ matrix.python-version }}`

Reply:
you might need to add something of the form:

```      - name: Install pipenv
        run: |
          python -m pip install --upgrade pipenv wheel
```

under steps, see e.g.: https://gist.github.com/alukach/6f3a371e9af600e417aca1b36806ad72